mlx/traceability.py

```diff
@@ -718,6 +754,32 @@ def process_item_nodes(app, doctree, fromdocname):
 
         node.replace_self(new_node)
 
+    # We process the item-link items. These are stored in a dictionary, so they
+    # can be looked up efficiently when traversing the list of Item objects below
+    # Structure of this dictionary is as follows:
+    #
+    # {
+    #   'SWRQT-lorem': [
+    #                   ('ITEST-ipsum', 'depends_on'),
+    #                   ('ITEST-dolor', 'verifies'),
+    #                  ]
+    #   'SWRQT-sit':   [
+    #                   ('ITEST-amet', 'depends_on'),
+    #                   ('ITEST-ipsum', 'depends_on'),
+    #                  ]
+    #
+    # }
+    itemlinks = {}
+    for node in doctree.traversingrse(ItemLink):
```

Comment:
this whole block needs to be moved up in the function, we first need to add all relations, before we render the item-list/matrix/tree/item...

Reply:
right before the self_test as we want these extra relations to be checked as well...